### PR TITLE
Feature slipstream

### PR DESCRIPTION
**Pull request by mclarke2**
This pull request is the first of four large pull request (Slipstream, Plots, VTOL and VLM) in that order.
Since there are significant changes in code structure please take some time to go through. These changes may not necessarily be all related to the branch but rather are related to all four branches.
The major changes are:
- Addition of a new aerodynamics method (Blown Wing Aero which does not use a surrogate, rather queries each point in the mission. NB. the propeller slipstream and its influence changes at each point in the mission
- Addition of propeller slipstream influence on the wing (Blown Wing VLM).
- Removal of "prop_attributes" from propeller data structure
- Update of propeller scripts to store data for VTOL and Noise analysis
- Removal of evtol weights methods since the vtol branch has not been pulled in. I am not sure how this passes since we have different networks for different vehicles
- Addition of Internal Combustion Engine

### CODE REVIEW

File: .coveragerc [hunk 1]
```diff
@@ -0,0 +1,28 @@
+[run]
```
Comment (WallyMaier) [inline .coveragerc]: what is this file?

File: regression/scripts/weights/eVTOL_Weights_Buildup_Regression.py [hunk 2]
```diff
@@ -1,364 +0,0 @@
-# eVTOL_Weights_Buildup_Regression.py
-#
-# Created: Feb, 2018, J. Smart
-
-#------------------------------------------------------------------------------
-# Imports
-#------------------------------------------------------------------------------
-
-import sys
```
Comment (WallyMaier) [inline regression/scripts/weights/eVTOL_Weights_Buildup_Regression.py]: wah, lets fix this ASAP

File: trunk/SUAVE/.idea/SUAVE.iml [hunk 3]
```diff
@@ -0,0 +1,11 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<module type="PYTHON_MODULE" version="4">
```
Comment (WallyMaier) [inline trunk/SUAVE/.idea/SUAVE.iml]: necessary?

File: trunk/SUAVE/.idea/misc.xml [hunk 4]
```diff
@@ -0,0 +1,4 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (WallyMaier) [inline trunk/SUAVE/.idea/misc.xml]: necessary?

File: trunk/SUAVE/.idea/modules.xml [hunk 5]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (WallyMaier) [inline trunk/SUAVE/.idea/modules.xml]: necessary?

File: trunk/SUAVE/.idea/workspace.xml [hunk 6]
```diff
@@ -0,0 +1,100 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project version="4">
```
Comment (WallyMaier) [inline trunk/SUAVE/.idea/workspace.xml]: necessary?

File: trunk/SUAVE/Analyses/Aerodynamics/Blown_Wing_Aero.py [hunk 7]
```diff
@@ -0,0 +1,106 @@
...
+class Blown_Wing_Aero(Markup):
+    """This is an analysis based on low-fidelity models. This model incorperates the blown effect of propellers mounted
+    in front of a wing.
+    separately
+
+    Assumptions:
+    Subsonic
+
+    Source:
+    Primarily based on adg.stanford.edu, see methods for details
```
Comment (WallyMaier) [inline trunk/SUAVE/Analyses/Aerodynamics/Blown_Wing_Aero.py]: possibly add the section

File: trunk/SUAVE/Analyses/Aerodynamics/Blown_Wing_Vortex_Lattice.py [hunk 8]
```diff
@@ -0,0 +1,135 @@
...
+# package imports
+import numpy as np
+
+
+# ----------------------------------------------------------------------
+#  Class
+# ----------------------------------------------------------------------
+## @ingroup Analyses-Aerodynamics
+class Blown_Wing_Vortex_Lattice(Aerodynamics):
+    """This model incorperate the propeller wake of propellers in the estimation of wing lift vortex lattice.
```
Comment (WallyMaier) [inline trunk/SUAVE/Analyses/Aerodynamics/Blown_Wing_Vortex_Lattice.py]: "incorporate"  
also, is the same source?

File: trunk/SUAVE/Analyses/Mission/Segments/Cruise/Constant_Speed_Constant_Altitude.py [hunk 9]
```diff
@@ -115,19 +115,19 @@ def __defaults__(self):
         
         # Update Conditions
         iterate.conditions = Process()
-        iterate.conditions.differentials   = Methods.Common.Numerics.update_differentials_time
+        iterate.conditions.differentials   = Methods.Common.Numerics.update_differentials_time
```
Comment (WallyMaier) [inline trunk/SUAVE/Analyses/Mission/Segments/Cruise/Constant_Speed_Constant_Altitude.py]: remove the spaces here

File: trunk/SUAVE/Analyses/Mission/Segments/Hover/Climb.py [hunk 10]
```diff
@@ -58,9 +58,11 @@ def __defaults__(self):
         # --------------------------------------------------------------
         #   User inputs
         # --------------------------------------------------------------
-        self.altitude_start = None # Optional
-        self.altitude_end   = 1. * Units.km
-        self.climb_rate     = 1.  * Units.m / Units.s
+        self.altitude_start     = None # Optional
+        self.altitude_end       = 1. * Units.km
+        self.thrust_angle_start = 90. * Units.degrees
+        self.thrust_angle_end   = 90. * Units['rad']
```
Comment (WallyMaier) [inline trunk/SUAVE/Analyses/Mission/Segments/Hover/Climb.py]: 90 rad????

File: trunk/SUAVE/Analyses/Mission/Segments/Transition/Constant_Acceleration_Constant_Pitchrate_Constant_Altitude.py [hunk 11]
```diff
@@ -0,0 +1,146 @@
+## @ingroup Analyses-Mission-Segments-Transition
+# Constant_Acceleration_Constant_Altitude.py
+#
+# Created:  Jan 2016, E. Botero
+# Modified: Jul 2017, T. MacDonald
```
Comment (WallyMaier) [inline trunk/SUAVE/Analyses/Mission/Segments/Transition/Constant_Acceleration_Constant_Pitchrate_Constant_Altitude.py]: where was this?   last updated 2017?

File: trunk/SUAVE/Components/Energy/Converters/Internal_Combustion_Engine.py [hunk 12]
```diff
@@ -0,0 +1,149 @@
...
+
+# package imports
+import numpy as np
+from SUAVE.Components.Energy.Energy_Component import Energy_Component
+
+# ----------------------------------------------------------------------
+#  Internal Combustion Engine Class
+# ----------------------------------------------------------------------
+
+class Internal_Combustion_Engine(Energy_Component):
```
Comment (WallyMaier) [inline trunk/SUAVE/Components/Energy/Converters/Internal_Combustion_Engine.py]: is the new default SUAVE style/structure?

File: trunk/SUAVE/Components/Energy/Networks/Internal_Combustion_Propeller.py [hunk 13]
```diff
@@ -0,0 +1,153 @@
...
+import numpy as np
+from SUAVE.Components.Propulsors.Propulsor import Propulsor
+from SUAVE.Core import Data, Units
+
+# ----------------------------------------------------------------------
+#  Network
+# ----------------------------------------------------------------------
+## @ingroup Components-Energy-Networks
+class Internal_Combustion_Propeller(Propulsor):
+    """ A simple mock up of an internal combustion propeller engine. Tis network adds an extra
```
Comment (WallyMaier) [inline trunk/SUAVE/Components/Energy/Networks/Internal_Combustion_Propeller.py]: So the difference between propeller and this is the unknowns?  Is there a better way to reduce both files into one? (thinking about user friendliness)

File: trunk/SUAVE/Components/Energy/Networks/Stopped_Rotor_Low_Fidelity.py [hunk 14]
```diff
@@ -0,0 +1,302 @@
+# Stopped_Rotor_Low_Fidelity.py
```
Comment (WallyMaier) [inline trunk/SUAVE/Components/Energy/Networks/Stopped_Rotor_Low_Fidelity.py]: what makes this file necessary compared to Stopped_Rotor.py?

File: trunk/SUAVE/Input_Output/OpenVSP/vsp_write.py [hunk 15]
```diff
@@ -691,7 +689,7 @@ def write_vsp_fuselage(fuselage,area_tags, main_wing, fuel_tank_set_ind):
         if 'z_pos' in vals.tail:
             tail_z_pos = vals.tail.z_pos
         else:
-            pass # use above default        
+            pass # use above default
```
Comment (WallyMaier) [inline trunk/SUAVE/Input_Output/OpenVSP/vsp_write.py]: remove space

File: trunk/SUAVE/Methods/Missions/Segments/Cruise/Constant_Mach_Linear_Altitude.py [hunk 16]
```diff
@@ -0,0 +1,49 @@
+import SUAVE
+
+
```
Comment (WallyMaier) [inline trunk/SUAVE/Methods/Missions/Segments/Cruise/Constant_Mach_Linear_Altitude.py]: white space

File: trunk/SUAVE/Methods/Missions/Segments/Cruise/Constant_Speed_Linear_Altitude.py [hunk 17]
```diff
@@ -0,0 +1,38 @@
+
+
```
Comment (WallyMaier) [inline trunk/SUAVE/Methods/Missions/Segments/Cruise/Constant_Speed_Linear_Altitude.py]: white space

File: trunk/SUAVE/Methods/Missions/Segments/Transition/Constant_Accelerated_Climb_Constant_Angle.py [hunk 18]
```diff
@@ -0,0 +1,163 @@
...
+    # Figure out vx
+    V  = (vf_mag-v0_mag) + v0_mag
+    vx = t_nondim *  V  * np.cos(climb_angle)
+    vz = t_nondim *  V  * np.sin(climb_angle)
+    
+    # set the body angle
+    body_angle =time*(Tf-T0)/(t_final-t_initial)
+    segment.state.conditions.frames.body.inertial_rotations[:,1] = body_angle[:,0]     
+
+
```
Comment (WallyMaier) [inline trunk/SUAVE/Methods/Missions/Segments/Transition/Constant_Accelerated_Climb_Constant_Angle.py]: white space

File: trunk/SUAVE/Methods/Weights/Buildups/Electric_Helicopter/empty.py [hunk 19]
```diff
@@ -74,9 +74,9 @@ def empty(config,
 # Unpack Inputs
 #-------------------------------------------------------------------------------
 
-    rProp               = config.propulsors.network.propeller.prop_attributes.tip_radius
-    mBattery            = config.propulsors.network.battery.mass_properties.mass
-    mPayload            = config.propulsors.network.payload.mass_properties.mass
+    rProp               = config.propulsors.propulsor.propeller.tip_radius
```
Comment (WallyMaier) [inline trunk/SUAVE/Methods/Weights/Buildups/Electric_Helicopter/empty.py]: why did this change to propulsors.propulsor?

File: trunk/SUAVE/Methods/Weights/Buildups/Electric_Stopped_Rotor/empty.py [hunk 20]
```diff
@@ -75,14 +75,15 @@ def empty(config,
...
+    rPropLift           = config.propulsors.propulsor.propeller_lift.tip_radius
+    rPropThrust         = config.propulsors.propulsor.propeller_forward.tip_radius
+    mBattery            = config.propulsors.propulsor.battery.mass_properties.mass
+    mPayload            = config.propulsors.propulsor.payload.mass_properties.mass
     MTOW                = config.mass_properties.max_takeoff
-    nLiftProps          = config.propulsors.network.number_of_engines/2
-    nThrustProps        = config.propulsors.network.number_of_engines/2
-    nLiftBlades         = config.propulsors.network.propeller.prop_attributes.number_blades
-    nThrustBlades       = config.propulsors.network.propeller.prop_attributes.number_blades
+    nLiftProps          = config.propulsors.propulsor.number_of_engines_lift
```
Comment (WallyMaier) [inline trunk/SUAVE/Methods/Weights/Buildups/Electric_Stopped_Rotor/empty.py]: remove the "/2"?  (just to to double check)

File: trunk/SUAVE/Methods/Weights/Buildups/Electric_Tiltrotor/empty.py [hunk 21]
```diff
@@ -116,19 +116,23 @@ def empty(config,
 
     # Component Weight Calculations
 
-    output.lift_rotors      = (prop(config.propulsors.network.propeller, maxLift)
+    output.lift_rotors      = (prop(config.propulsors.propulsor.propeller, maxLift)
                                * (len(config.wings['main_wing'].motor_spanwise_locations) 
-                                  + len(config.wings['main_wing'].motor_spanwise_locations)))
+                                  + len(config.wings['main_wing_2'].motor_spanwise_locations))) # make more generic ash jordan about this
```
Comment (WallyMaier) [inline trunk/SUAVE/Methods/Weights/Buildups/Electric_Tiltrotor/empty.py]: can this be done before the pull?

File: trunk/SUAVE/Methods/Weights/Buildups/Electric_Helicopter/empty.py (hunk 19 above)
Comment (JTrentSmart) [reply trunk/SUAVE/Methods/Weights/Buildups/Electric_Helicopter/empty.py]: Depending on which energy network is being used by the aircraft, the call to the energy network is different. Some inherit straight from the network class and thus need propulsors.network as the call and some need propulsors.propulsor. It's one of the things I'd like to address is a separate pull request I'm going to be presenting about shortly.

File: regression/scripts/weights/eVTOL_Weights_Buildup_Regression.py (hunk 2 above)
Comment (mclarke2) [reply regression/scripts/weights/eVTOL_Weights_Buildup_Regression.py]: It's coming back in when I pull in the VTOL branch

File: trunk/SUAVE/.idea/SUAVE.iml (hunk 3 above)
Comment (mclarke2) [reply trunk/SUAVE/.idea/SUAVE.iml]: I'm not sure where these changes came from. I think its part of the new python we are pulling in

File: trunk/SUAVE/.idea/misc.xml (hunk 4 above)
Comment (mclarke2) [reply trunk/SUAVE/.idea/misc.xml]: same comment as above

File: trunk/SUAVE/.idea/modules.xml (hunk 5 above)
Comment (mclarke2) [reply trunk/SUAVE/.idea/modules.xml]: same comment as above

File: trunk/SUAVE/Methods/Missions/Segments/Cruise/Constant_Speed_Linear_Altitude.py (hunk 17 above)
Comment (mclarke2) [reply trunk/SUAVE/Methods/Missions/Segments/Cruise/Constant_Speed_Linear_Altitude.py]: corrected

File: trunk/SUAVE/Methods/Missions/Segments/Transition/Constant_Accelerated_Climb_Constant_Angle.py (hunk 18 above)
Comment (mclarke2) [reply trunk/SUAVE/Methods/Missions/Segments/Transition/Constant_Accelerated_Climb_Constant_Angle.py]: corrected, I needed one white space for aesthetics

File: trunk/SUAVE/Methods/Weights/Buildups/Electric_Helicopter/empty.py (hunk 19 above)
Comment (mclarke2) [reply trunk/SUAVE/Methods/Weights/Buildups/Electric_Helicopter/empty.py]: This is going to be corrected in the VTOL branch which will be soon pulled into peudo develop

File: trunk/SUAVE/Methods/Weights/Buildups/Electric_Stopped_Rotor/empty.py (hunk 20 above)
Comment (mclarke2) [reply trunk/SUAVE/Methods/Weights/Buildups/Electric_Stopped_Rotor/empty.py]: refer to above comment

File: trunk/SUAVE/Methods/Weights/Buildups/Electric_Tiltrotor/empty.py (hunk 21 above)
Comment (mclarke2) [reply trunk/SUAVE/Methods/Weights/Buildups/Electric_Tiltrotor/empty.py]: refer to above comment

File: .coveragerc (hunk 1 above)
Comment (mclarke2) [reply .coveragerc]: Not sure, this is something for coverage not suave itself

**Pull request merged**